ibis/expr/window.py

```diff
@@ -12,16 +15,11 @@ def _sequence_to_tuple(x):
     return tuple(x) if util.is_iterable(x) else x
 
 
-def _determine_how(preceding):
-    if isinstance(preceding, tuple):
-        start, end = preceding
-        if start is None:
-            offset_type = type(end)
-        else:
-            offset_type = type(start)
-    else:
-        offset_type = type(preceding)
+RowsWithMaxLookback = namedtuple('RowsWithMaxLookback', 'rows max_lookback')
```

Comment:
Let's use [`typing.NamedTuple`](https://docs.python.org/3/library/typing.html#typing.NamedTuple) here, to allow for easier type annotations in the future. We can't use the variable annotation syntax yet (it's available from Python 3.6 onward) so we'll use the equivalent example that's just below the first.

Reply:
Changed to typing.NamedTuple in the next commit